receiver/prometheusreceiver/internal/structhash.go

```diff
@@ -0,0 +1,264 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
Do we really need this whole hashing implementation just for this? Was there a problem with the implementation from the operator itseif? That one was much simpler.

Reply:
@swiatekm-sumo - Please see this comment - https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30258#pullrequestreview-1817059446.
There was a feedback about sorting, so that approach wouldn't work well unless we made changes. So I used the OSS library which implemented this already.

Reply:
Why can't we just sort the keys in that loop? goyaml is otherwise deterministic in its outputs.

Reply:
@swiatekm-sumo - Thanks for reviewing. Yes, that could be done as well. Since there was an implementation already, I picked this library since it was already doing it and we discussed this in the SIG meeting a couple weeks ago. If you and @dashpole/@Aneurysm9 agree I can change the implementation to do that instead.

Reply:
If the maintainers of this component are ok with maintaining this hashing code specifically for this purpose, I guess I have no objections to that.

Reply:
Thanks @swiatekm-sumo. @dashpole @Aneurysm9 - Please let me know your thoughts. Thanks!